packs/salt/actions/lib/base.py

```diff
@@ -27,15 +41,24 @@ class SaltAction(Action):
     def __init__(self, config):
         super(SaltAction, self).__init__(config=config)
         self.url = self.config.get('api_url', None)
+        self.eauth = self.config.get('eauth', None)
         self.username = self.config.get('username', None)
         self.password = self.config.get('password', None)
 
-    def generate_package(self, client='local', cmd=None):
+    def generate_package(self, client='local', cmd=None, **kwargs):
         self.data = SaltPackage(client).data
+        self.data['eauth'] = self.eauth
         self.data['username'] = self.username
         self.data['password'] = self.password
         if cmd:
             self.data['fun'] = cmd
+        if client is 'local':
+            self.data['tgt'] = kwargs.get('target', '*')
+            self.data['expr_form'] = kwargs.get('expr_form', 'glob')
+        if kwargs['kwargs'] is not None:
+            self.data['kwarg'] = kwargs['kwargs']['kwargs']
```

Comment:
Actually it looks like it is. All this just to bring out the nested `kwargs` and rename it to `kwarg`. Laugh away! :laughing: